packages/clay-navigation/src/Vertical.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import ClayIcon from '@clayui/icon';
+import React from 'react';
+import {Nav} from './Nav';
+import {NavItem} from './NavItem';
+import {NavLink} from './NavLink';
+
+interface IItemProps {
+	active?: boolean;
+	onClick?: () => void;
+	href?: string;
+	label?: string;
+}
+
+interface IItem extends IItemProps {
+	collapsed?: boolean;
+	items?: Array<IItemProps>;
+}
+
+interface IProps {
+	activeLabel: string;
+	items: Array<IItem>;
+	large?: boolean;
+	spritemap?: string;
+}
+
+function renderItems(
+	items: Array<IItem>,
+	collapsedMap: any,
+	setCollapsedMap: any,
+	spritemap?: string,
+	level = 0
+) {
+	return items.map(({active, href, items: subItems, label}, i) => {
+		const key = `${level}-${i}`;
+		const collapsed = collapsedMap[key];
+
+		return (
+			<NavItem key={i}>
+				<NavLink
```

Comment:
Chances are we also need to use a `ClayButton` here for when people want to control the component instead of creating link navigation.